pkg/controller/complianceremediation/complianceremediation_controller.go

```diff
@@ -369,6 +379,51 @@ func (r *ReconcileComplianceRemediation) countKubeUnmetDependencies(rem *compv1a
 	return nMissingDeps, nil
 }
 
+func (r *ReconcileComplianceRemediation) handleUnsetValues(rem *compv1alpha1.ComplianceRemediation, logger logr.Logger) (reconcile.Result, error) {
+	_, hasUnsetXccdValues := rem.Annotations[compv1alpha1.RemediationUnsetValueAnnotation]
+
+	var nNotSetValues int
+	if hasUnsetXccdValues {
+		var xccdfValueErr error
+		nNotSetValues, xccdfValueErr = r.countXCCDFUnsetValues(rem, logger)
+		if xccdfValueErr != nil {
+			return reconcile.Result{}, xccdfValueErr
+		}
+	} else {
+		return reconcile.Result{}, fmt.Errorf("Remediation has RemediationUnsetValueAnnotation but no un-set XCCDF values found")
+	}
+
+	rCopy := rem.DeepCopy()
+	labels := rCopy.GetLabels()
+	if labels == nil {
+		labels = make(map[string]string)
+	}
+
+	if nNotSetValues > 0 {
+		if _, ok := labels[compv1alpha1.RemediationUnsetValueLabel]; !ok {
+			labels[compv1alpha1.RemediationUnsetValueLabel] = ""
+			rCopy.SetLabels(labels)
+			err := r.client.Update(context.TODO(), rCopy)
+			if err != nil {
+				return reconcile.Result{}, fmt.Errorf("adding un-set xccdf value label: %w", err)
+			}
+		}
+		return reconcile.Result{}, nil
+	}
+	return reconcile.Result{}, nil
+}
+
+func (r *ReconcileComplianceRemediation) countXCCDFUnsetValues(rem *compv1alpha1.ComplianceRemediation, logger logr.Logger) (int, error) {
+	var nNotSetValues int
+	not_set_values := rem.Annotations[compv1alpha1.RemediationUnsetValueAnnotation]
```

Comment:
Rather than use underscores for a similarly named variable, it would be better to have different names, (such as numberUnset and unsetValues)

Reply:
Yeah, that's a good idea